target/ppc/fpu_helper.c

```diff
@@ -2179,23 +2182,23 @@ void helper_##op(CPUPPCState *env, ppc_vsr_t *xt,                             \
     vex_flag = fpscr_ve && (vxvc_flag || vxsnan_flag);                        \
                                                                               \
     if (!vex_flag) {                                                          \
-        if (float64_##cmp(xb->VsrD(0), xa->VsrD(0),                           \
+        if (tp##_##cmp(xb->fld, xa->fld,                                      \
                           &env->fp_status) == exp) {                          \
-            t.VsrD(0) = -1;                                                   \
-            t.VsrD(1) = 0;                                                    \
+            t.f128 = make_float128(-1ULL, sdwr);                              \
```

Comment:
.... or initialize `t` with zero, use `memset` here, and drop the `else` branch.

Reply:
I went with this idea. Even tough getting rid of a variable is good, I think removing the extra parameter makes the code more readable and might be more interesting than the possible (which I think would be small) gain in performance.

Thank you again :)